lib/connection.js

```diff
@@ -26,7 +27,16 @@ function Connection(options) {
 
     this.queue = {};
 
-    let statsdClient = new StatsD();
+    let statsdUrl = process.env.STATSD_URL && url.parse(process.env.STATSD_URL);
+    let statsdClient = new StatsD(
+    statsdUrl
+        ? {
+            host: statsdUrl.hostname,
+            port: statsdUrl.port,
+            cacheDns: true,
+        }
+        : {}
+    );
     this.metricReporter = new MetricReporter(statsdClient, null);
```

Comment:
I don't believe we want a different reporter for each connection, client, and producer. Having multiple is unnecessary and could cause us headaches if we share metric names across them, particularly for gauges.